### ISSUE TITLE
todo-tree produced no results; error "Running the contributed command: 'todo-tree.refresh' failed."

### ISSUE THREAD

**Issue by mdrayman** [Gruntfuggly/todo-tree#544]
todo-tree shows no results on my python project I'm working on.  I tested other languages and it has the same failure.  When I manually click refresh button, vscode pops up an error on lower right saying "Running the contributed command: 'todo-tree.refresh' failed."  I read entire bug listing here, https://github.com/Gruntfuggly/todo-tree/issues/253, but that didn't help me at all.  I have latest vscode, Version 1.59.1 (2021-08-19T11:56:46.957Z).  I also tried uninstalling and reinstalling todo-tree without success.  I only changed a couple settings for todo-tree, but these were changed after i'd noticed this failure.  These are all my todo-tree settings from settings.json:

    "todo-tree.regex": "((--|//|#|<!--|;|/\\*|^)\\s*($TAGS)|^\\s*- \\[ \\])",
    "todo-tree.regex.regex": "((--|//|#|<!--|;|/\\*|^)\\s*($TAGS)|^\\s*- \\[ \\])",
    "todo-tree.tree.showScanModeButton": false,
    "todo-tree.general.debug": true,
    "todo-tree.general.enableFileWatcher": true,
    "todo-tree.general.statusBar": "total"

**Comment by Gruntfuggly** [comment]
It's because of the first setting - there is no setting `todo-tree.regex`.

However, I don't quite understand why it causes that error. It's probably because it is not able to read the sub-properties. I will have a look into it.

**Comment by ibraimgm** [comment]
@mdrayman I just  stumbled on this problem today, and to see the "true" error, I had to disable the extension, reload vscode, and enable it. In my case, the problem was that the extension somehow wasn't able to find `ripgrep`, so I just added `"todo-tree.ripgrep.ripgrep": "/absolute/path/to/ripgrep"` to my vscode settings, disabled and reenabled the extension again and it was fixed.

Be mindful to use the **absolute** path to the ripgrep executable, even if it is available on your shell `$PATH`.
